core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -978,17 +978,17 @@ private void cleanupExpiredRemoteLogSegments() throws RemoteStorageException, Ex
                     }
                     RemoteLogSegmentMetadata metadata = segmentsIterator.next();
 
-                    // check whether the segment contains the required epoch range with in the current leader epoch lineage.
-                    boolean isValidSegment = isRemoteSegmentWithinLeaderEpochs(metadata, logEndOffset, epochWithOffsets);
-                    boolean isSegmentDeleted = false;
-                    if (isValidSegment) {
-                        isSegmentDeleted =
-                                remoteLogRetentionHandler.deleteRetentionTimeBreachedSegments(metadata) ||
-                                        remoteLogRetentionHandler.deleteRetentionSizeBreachedSegments(metadata);
-                    }
+                    boolean isSegmentDeleted = remoteLogRetentionHandler.deleteLogStartOffsetBreachedSegments(
+                            metadata, logStartOffset, epochWithOffsets);
```

Comment:
Thanks for the update. The logic makes sense now. Could you please add some comment here to explain why we need to separate the `deleteLogStartOffsetBreachedSegments` here? Thanks.